guide_drafts/migration_guide.md

```diff
@@ -0,0 +1,277 @@
+# Diesel 2.0 Migration guide
+
+Diesel 2.0 changes large parts of the internal implementation of diesel. 
+In some cases this impacts code written using Diesel 1.4.x. 
+This document outlines notable changes and presents potential update strategies. 
+We recommend to start the upgrade by removing the usage of all items that 
+are deprecated on Diesel 1.4.x.
+
+We expect that not all users are effected equally by different required changes. 
```

Comment:
I am not sure this lines is necessary, especially since the wording of the next one specifically states that "your mileage may vary", basically